### PR TITLE
operator/ingress: Add HTTP cookie capture

### PR DESCRIPTION
**Pull request by Miciah**
* `operator/v1/types_ingress.go` (`CookieMatchType`): New type.  Specify a type of match (exact or prefix) for identifying cookies.
* `operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml-merge-patch`: Patch the validation of the `spec.logging.access.httpCaptureCookies` field to require the `name` subfield if the `matchType` subfield is "Exact" and require the `namePrefix` subfield if `matchType` is "Prefix".
(`IngressControllerCaptureHTTPCookies`): New type.  Specify a cookie name, maximum value length, and match type (using the new `CookieMatchType` type) to use when capturing cookies for access logs.
(`AccessLogging`): Add `HTTPCaptureCookies` field of type `IngressControllerCaptureHTTPCookies`.
* `operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml`: Regenerate.

----

Depends on https://github.com/openshift/api/pull/686.

### CODE REVIEW

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -483,6 +483,84 @@ type LoggingDestination struct {
...
+// IngressControllerCaptureHTTPCookies specifies which HTTP cookies the
+// IngressController captures.
+type IngressControllerCaptureHTTPCookies struct {
+	// name specifies a cookie name.  Its value must be a valid HTTP cookie
+	// name as defined in RFC 6265 section 4.1
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[\x21\x23-\x27\x2a\x2b\x2d\x2e\x30-\x39\x41-\x5a\x5e-\x7a\x7c\x7e]+$`
+	// +required
+	Name string `json:"name"`
```
Comment (deads2k) [inline operator/v1/types_ingress.go:543]: NamePrefix?

File: operator/v1/types_ingress.go [hunk 2]
```diff
@@ -501,6 +579,18 @@ type AccessLogging struct {
 	//
 	// +optional
 	HttpLogFormat string `json:"httpLogFormat,omitempty"`
+
+	// httpCaptureHeaders defines HTTP headers that should be captured in
+	// access logs.  If this field is empty, no headers are captured.
+	//
+	// +optional
+	HTTPCaptureHeaders *IngressControllerCaptureHTTPHeaders `json:"httpCaptureHeaders,omitempty"`
```
Comment (deads2k) [inline operator/v1/types_ingress.go:587]: I don't see why this one should be a pointer. The slices can just be zero length and it does the same thing, right?

File: operator/v1/types_ingress.go (hunk 1 above)
Comment (deads2k) [reply operator/v1/types_ingress.go:543]: > NamePrefix?

Oh, I see below, this is a union struct then, right?

Comment (Miciah) [reply operator/v1/types_ingress.go:667]: I'll make it a union.

File: operator/v1/types_ingress.go (hunk 2 above)
Comment (Miciah) [reply operator/v1/types_ingress.go]: Right, that would work.  Updated.

**Comment by Miciah** [comment]
Rebased.  #686 merged, so this PR is unblocked.

**Comment by Miciah** [comment]
Added more validation.

**Pull request merged**